### PR TITLE
OD-596 [Feature] Added app version number validation

### PR DESCRIPTION
**Pull request by skrupnyk**
@romanyosyfiv

OD-596 https://weboo.atlassian.net/browse/OD-596

## Description
Added app version number validation. The user will not be able to submit the form if the version number is in the wrong format or is less than a pre-existing version number.

## Screenshots/screencasts
https://storyxpress.co/video/kvuqi9hn52bid7iwn

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: js/interface.js [hunk 1]
```diff
@@ -2964,6 +2970,54 @@ $('#appStoreConfiguration, #enterpriseConfiguration, #unsignedConfiguration').on
...
+      }
+
+      var segmentedOldVersion = previosVersion.split('.');
+      var segmentedNewVersion = newVersion.split('.');
+
+      for (var i = 0; i < segmentedNewVersion.length; i++) {
+        var a = parseInt(segmentedNewVersion[i], 10) || 0;
+        var b = parseInt(segmentedOldVersion[i], 10) || 0;
+
+        if (a > b) {
```
[the remark this answers is not in the document]
Comment (skrupnyk) [reply js/interface.js:2991]: > What happens if `a` and `b` are the same?

In this case, the cycle will end and the same error message will appear. You can see it below in the code.

File: js/interface.js [hunk 2]
```diff
@@ -2964,6 +2970,54 @@ $('#appStoreConfiguration, #enterpriseConfiguration, #unsignedConfiguration').on
   Fliplet.Widget.autosize();
 });
 
+$('form').validator({
+  custom: {
+    'validation-version-number': function($el) {
+      var previosVersion = $el.data('validation-version-number');
```
[the remark this answers is not in the document]
Comment (skrupnyk) [reply js/interface.js]: > Keeping terminologies consistent.

fixed

File: js/interface.js [hunk 3]
```diff
@@ -2964,6 +2970,54 @@ $('#appStoreConfiguration, #enterpriseConfiguration, #unsignedConfiguration').on
...
+      for (var i = 0; i < segmentedNewVersion.length; i++) {
+        var a = parseInt(segmentedNewVersion[i], 10) || 0;
+        var b = parseInt(segmentedOldVersion[i], 10) || 0;
+
+        if (a > b) {
+          return false;
+        }
+
+        if (a < b) {
+          $el.attr('data-validation-version-number-error', 'Please make sure the version number is higher than ' + $el.data('validation-version-number'));
```
[the remark this answers is not in the document]
Comment (skrupnyk) [reply js/interface.js]: > You shouldn't need to read the value again.

fixed

**Review by tonytlwu — APPROVED** [review]
@romanyosyfiv @inna-bieshulia This is ready for testing

Risk: Medium
Affected: Studio users submitting iOS app builds
Verify: App builds are submitted with the correct version numbers

Squash and merge: Use PR title

**Pull request merged**